Objects/typeobject.c

```diff
@@ -1312,16 +1312,21 @@ subtype_traverse(PyObject *self, visitproc visit, void *arg)
         assert(base);
     }
 
-    if (type->tp_flags & Py_TPFLAGS_MANAGED_DICT) {
-        int err = _PyObject_VisitManagedDict(self, visit, arg);
-        if (err) {
-            return err;
+    if (type->tp_dictoffset != base->tp_dictoffset) {
+        assert(base->tp_dictoffset == 0);
```

Comment:
Yes. Either `type->tp_dictoffset` is the same as `base->tp_dictoffset` because it is inherited, *or* the base class had no dict, and `base->tp_dictoffset == 0`.